### PR TITLE
update to new cogs-client API, remove useAudioPlayer

### PR DESCRIPTION
**Pull request by chetbox**
...﻿in favour of CogsAudioPlayer from cogs-client

Depends on https://github.com/clockwork-dog/cogs-client-lib/pull/10

### CODE REVIEW

File: src/components/Background.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        top: 0,
+        bottom: 0,
+        left: 0,
+        right: 0,
+        backgroundColor: background?.color || 'transparent',
+        backgroundImage: background?.image?.file ? `url("${background.image.file}")` : 'none',
+        backgroundRepeat: 'no-repeat',
+        backgroundPosition: 'center',
+        backgroundSize: background?.image?.fit ?? 'auto',
+        transition: 'filter 200ms',
```
Comment (tomdixon) [inline src/components/Background.tsx:31]: Needed?

Comment (chetbox) [reply src/components/Background.tsx]: Removed.

File: src/hooks/useBackground.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { useCallback, useState } from 'react';
+import useCogsMessage from './useCogsMessage';
+
+export default function useBackground(connection: CogsConnection): BackgroundOptions | null {
+  const [background, setBackground] = useState<BackgroundOptions | null>(null);
+
+  useCogsMessage(
+    connection,
+    useCallback((message) => {
+      switch (message.type) {
```
Comment (tomdixon) [inline src/hooks/useBackground.ts:11]: Simple `if` here ?

File: src/hooks/useHint.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-        break;
-    }
-  }, []);
-
-  const callbacks = useMemo((): Callbacks => ({ onMessage }), [onMessage]);
-  useCogsCallbacks(connection, callbacks);
+  useCogsMessage(
+    connection,
+    useCallback((message) => {
+      switch (message.type) {
```
Comment (tomdixon) [inline src/hooks/useHint.ts:11]: Again, `if`?

File: src/hooks/useBackground.ts (hunk 2 above)
Comment (chetbox) [reply src/hooks/useBackground.ts]: Done.

File: src/hooks/useHint.ts (hunk 3 above)
Comment (chetbox) [reply src/hooks/useHint.ts]: Done.

**Review by tomdixon — APPROVED** [review]

**Pull request merged**